src/cephadm/cephadm.py

```diff
@@ -301,6 +302,52 @@ def get_daemon_args(self) -> List[str]:
                 r += ['--default-mon-cluster-log-to-stderr=true']
         return r
 
+    @staticmethod
+    def get_ceph_mounts(
+        ctx: CephadmContext,
+        ident: DaemonIdentity,
+        no_config: bool = False,
+    ) -> Dict[str, str]:
+        # Warning: This is a hack done for more expedient refactoring
+        mounts = _get_container_mounts_for_type(
+            ctx, ident.fsid, ident.daemon_type
+        )
+        data_dir = ident.data_dir(ctx.data_dir)
+        if ident.daemon_type == 'rgw':
+            cdata_dir = '/var/lib/ceph/radosgw/ceph-rgw.%s' % (
+                ident.daemon_id
+            )
+        else:
+            cdata_dir = '/var/lib/ceph/%s/ceph-%s' % (
+                ident.daemon_type,
+                ident.daemon_id,
+            )
+        if ident.daemon_type != 'crash':
+            mounts[data_dir] = cdata_dir + ':z'
+        if not no_config:
+            mounts[data_dir + '/config'] = '/etc/ceph/ceph.conf:z'
+        if ident.daemon_type in [
+            'rbd-mirror',
+            'cephfs-mirror',
+            'crash',
+            'ceph-exporter',
+        ]:
+            # these do not search for their keyrings in a data directory
+            mounts[
+                data_dir + '/keyring'
+            ] = '/etc/ceph/ceph.client.%s.%s.keyring' % (
+                ident.daemon_type,
+                ident.daemon_id,
+            )
```

Comment:
is this a black format thing? I find this particular little block a bit harder to read than the original line (even if it was overly long). If it's what the tool does though I can live with it.

Reply:
Yes. It is a bit unnatural. I think a quick fix would be to move `                data_dir + '/keyring'` to the line above via a variable assignment, and then black ought to make the block less weird looking. I can do that here in this PR or later on in a follow up series. Let me know what you prefer

Reply:
I don't want to hold off merging this over the formatting, so it can be done in a follow up